File: py/leetcode_easy/palindrome_number.py
def isPalindrome(x):
    if x < 0 or (x % 10 == 0 and x != 0):
        return False

    if x < 10:
        return True   

    reversed = 0
    while x > reversed:
        reversed = reversed * 10 + x % 10
        x = int(x / 10)
        
    return x == reversed or x == int(reversed / 10) 

File: go/leetcode_easy/palindrome_number/palindrome_number.go
package palindrome_number

// Palindrome Number
// https://leetcode.com/problems/palindrome-number/

func IsPalindrome(x int) bool {
	if x < 0 || (x%10 == 0 && x != 0) {
		return false
	}

	if x < 10 {
		return true
	}

	reversed := 0

	for x > reversed {
		reversed = reversed*10 + x%10
		x = x / 10
	}

	return x == reversed || x == reversed/10
}
